### PR TITLE
Move to contrib installation of spack-stack on Jet

### LINKED ISSUE #2841: Migrate Jet to /lfs5
### What new functionality do you need?

The /lfs4 filesystem has become unusable and users need to migrate to /lfs5. The global-workflow, libraries, and components (both internal and external) will need to be updated to use /lfs5.

### What are the requirements for the new functionality?

The following need to be updated to use /lfs5:

- [x] spack-stack (need install of v1.6.0 on /lfs5) - `/contrib/spack-stack/spack-stack-1.6.0/envs/gsi-addon-intel/install/modulefiles/Core`
- [ ] global-workflow
- [x] fix/`FIX_DIR` - relocated to `/lfs5/HFIP/hfv3gfs/glopara/FIX/fix`

The following need to be updated to use the migrated spack-stack install before global-workflow can be fully migrated:

- [ ] gdas (?)
- [ ] gfs-utils
- [ ] gsi-enkf
- [ ] gsi-monitor
- [ ] gsi-utils
- [ ] UPP
- [ ] UFS_UTILS
- [ ] ufs-weather-model
- [ ] EMC_verif-global
- [ ] wxflow (?)
- [x] obsproc - `obsproc.v1.2.0-rd-gfsv17` tag cut and installed everywhere (Jet: `/lfs5/HFIP/hfv3gfs/glopara/git/obsproc/v1.2.0`)
- [x] prepobs - `prepobs.v1.1.0-rd-gfsv17` tag cut and installed everywhere (Jet: `/lfs5/HFIP/hfv3gfs/glopara/git/prepobs/v1.1.0`)
- [x] Fit2Obs - `v1.1.3` tag cut and installed everywhere (Jet: `/lfs5/HFIP/hfv3gfs/glopara/git/Fit2Obs/v1.1.3`)
- [ ] tracker

### Acceptance Criteria

All components build and run on /lfs5

### Suggest a solution (optional)

_No response_

### PR DESCRIPTION
**Pull request by InnocentSouopgui-NOAA**
# Description
Migrates Global Workflow to use contrib installation of spack-stack  on Jet.
Following the failure of the storage /lfs4 on Jet, the installation of spack spack moved to /contrib.
All softwares relying on spack-stack on Jet needs update.

  Resolves #2841 
  Refs NOAA-EMC/gfs-utils#78
  Refs NOAA-EMC/GSI#786
  Refs NOAA-EMC/GSI-Monitor#143
  Refs NOAA-EMC/GSI-utils#51
  Refs ufs-community/UFS_UTILS#977

# Type of change
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES/NO (If YES, please add a link to any PRs that are pending.)
  - [ ] EMC verif-global
  - [ ] GDAS
  - [X] GFS-utils NOAA-EMC/gfs-utils#79
  - [X] GSI NOAA-EMC/GSI#787
  - [X] GSI-monitor NOAA-EMC/GSI-Monitor#144
  - [X] GSI-utils NOAA-EMC/GSI-utils#52
  - [X] UFS-utils ufs-community/UFS_UTILS#978
  - [ ] UFS-weather-model
  - [ ] wxflow

# How has this been tested?
Example:
- Clone and build on Jet
- Cycled experiments (48+ hours) at resolutions
  - [x] 96/48 on kjet
  - [x] 192/96 on kjet
  - [x] 384/192 on kjet

- Forecast only experiment (48+ hours) at resolutions
  - [ ] 48
  - [ ] 96
  - [ ] 192
  - [ ] 384

# Checklist
- [ ] Any dependent changes have been merged and published
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] New and existing tests pass with my changes
- [X] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by InnocentSouopgui-NOAA** [comment]
I am getting errors of the form bellow in forecast steps (tasks gdasfcst_seg0 and all enkfgdasfcst_mem###)
That is while running C192/C96 on Jet. It happened from the very first cycle, so it did not complete a single cycle.

@DavidHuber-NOAA 

```
21: Warn_K=   6 (i,j)=   87   12 (lon,lat)=123.209 -43.765 VA = 264.64157
21:      K=   5    338.73022
21:      K=   7    217.40834
21: Warn_K=   6 (i,j)=   88   13 (lon,lat)=121.423 -43.705 VA = 250.69765
21:      K=   5    297.46832
21:      K=   7    213.49741
21: Warn_K=   6 (i,j)=   84   16 (lon,lat)=122.124 -46.976 VA = 251.07759
21:      K=   5    256.00562
21:      K=   7    241.60887
 0: PASS: fcstRUN phase 2, n_atmsteps =               27 time is         1.264142
 5:
 5: FATAL from PE     5: NaN in input field of mpp_reproducing_sum(_2d), this indicates numerical instability
 5:
13:
13: FATAL from PE    13: NaN in input field of mpp_reproducing_sum(_2d), this indicates numerical instability
13:
21:
21: FATAL from PE    21: NaN in input field of mpp_reproducing_sum(_2d), this indicates numerical instability
21:
13: Image              PC                Routine            Line        Source
13: ufs_model.x        00000000086C51A7  Unknown               Unknown  Unknown
13: ufs_model.x        00000000078AD1B9  mpp_mod_mp_mpp_er          72  mpp_util_mpi.inc
13: ufs_model.x        0000000007B61AB6  mpp_efp_mod_mp_mp         195  mpp_efp.F90
13: ufs_model.x        0000000007AB2D99  mpp_domains_mod_m         143  mpp_global_sum.fh
13: ufs_model.x        0000000003E9EB6A  fv_grid_utils_mod        3077  fv_grid_utils.F90
13: ufs_model.x        0000000003F2ED3E  fv_mapz_mod_mp_la         794  fv_mapz.F90
13: libiomp5.so        0000146B48A6CBB3  __kmp_invoke_micr     Unknown  Unknown
13: libiomp5.so        0000146B489E8FAC  __kmp_fork_call       Unknown  Unknown
13: libiomp5.so        0000146B489AACB5  __kmpc_fork_call      Unknown  Unknown
13: ufs_model.x        0000000003F2A129  fv_mapz_mod_mp_la         683  fv_mapz.F90
13: ufs_model.x        0000000003E2EE61  fv_dynamics_mod_m         771  fv_dynamics.F90
13: ufs_model.x        0000000003C9236C  atmosphere_mod_mp         688  atmosphere.F90
13: ufs_model.x        0000000003A3490D  atmos_model_mod_m         879  atmos_model.F90
13: ufs_model.x        00000000035F688C  module_fcst_grid_        1335  module_fcst_grid_comp.F90
```

**Comment by InnocentSouopgui-NOAA** [comment]
It seems to be just a bad day

**Comment by InnocentSouopgui-NOAA** [comment]
I build initial conditions for other days and it cycled smoothly.

**Comment by RussTreadon-NOAA** [comment]
GSI PR [#787](https://github.com/NOAA-EMC/GSI/pull/787) has been merged into GSI `develop`.  Done at [`9f44c87`](https://github.com/NOAA-EMC/GSI/commit/9f44c8798c2087aca06df8f629699632e57df431).

The `sorc/gsi_enkf.fd` hash in `InnocentSouopgui-NOAA:migration-jet-contrib` must be updated to [`9f44c87`](https://github.com/NOAA-EMC/GSI/commit/9f44c8798c2087aca06df8f629699632e57df431) to bring these changes into g-w.

**Comment by InnocentSouopgui-NOAA** [comment]
A check is failing with the following message,
```
fatal: Fetched in submodule path 'ufs_utils.fd', but it did not contain 0426bf793051530794ec8f182e04f5cf129d0a90. Direct fetching of that commit failed.
```
How to diagnose what is going on?

**Comment by DavidHuber-NOAA** [comment]
@InnocentSouopgui-NOAA I suspect that the hash you are pointing to is for your own branch.  Update the hash instead to https://github.com/ufs-community/UFS_UTILS/commit/06eec5b6f636123835e2dfd9fc5229980c006735.

**Comment by InnocentSouopgui-NOAA** [comment]
> @InnocentSouopgui-NOAA I suspect that the hash you are pointing to is for your own branch. Update the hash instead to [ufs-community/UFS_UTILS@06eec5b](https://github.com/ufs-community/UFS_UTILS/commit/06eec5b6f636123835e2dfd9fc5229980c006735).

Thanks @DavidHuber-NOAA , that was the issue.

**Comment by DavidHuber-NOAA** [comment]
@InnocentSouopgui-NOAA Just a heads up, there is a bug in the newest GSI-utils that will cause the gdasanalcalc job to fail when performing GDASApp analyses (i.e. the `C96C48_ufs_hybatmDA` CI test) as noted in https://github.com/NOAA-EMC/global-workflow/pull/2819#issuecomment-2340706398.

**Comment by InnocentSouopgui-NOAA** [comment]
@DavidHuber-NOAA , should delay this PR for that bug?
I updated all the hash, I am doing a final test at res C96/48 before getting this out for review.

**Comment by DavidHuber-NOAA** [comment]
I think a delay would be preferred, unfortunately, as this will break development for GDASApp.  That said, I think we can go ahead and review it.

**Comment by InnocentSouopgui-NOAA** [comment]
Everything is ready for review.

File: ush/forecast_predet.sh [hunk 1]
```diff
@@ -548,6 +548,13 @@ FV3_predet(){
       ${NCP} "${PARMgfs}/post/gefs/postxconfig-NT-gefs-f00.txt"   "${DATA}/postxconfig-NT_FH00.txt"
     fi
   fi
+#  # before
+#  if [[ "${WRITE_DOPOST:-}" == ".true." ]]; then
+#    ${NCP} "${PARMgfs}/post/post_tag_gfs${LEVS}"                              "${DATA}/itag"
+#    ${NCP} "${FLTFILEGFS:-${PARMgfs}/post/postxconfig-NT-GFS-TWO.txt}"        "${DATA}/postxconfig-NT.txt"
+#    ${NCP} "${FLTFILEGFSF00:-${PARMgfs}/post/postxconfig-NT-GFS-F00-TWO.txt}" "${DATA}/postxconfig-NT_FH00.txt"
+#    ${NCP} "${POSTGRB2TBL:-${PARMgfs}/post/params_grib2_tbl_new}"             "${DATA}/params_grib2_tbl_new"
+#  fi
```
Comment (WalterKolczynski-NOAA) [inline ush/forecast_predet.sh]: Please remove this commented code.

File: sorc/gdas.cd
Comment (WalterKolczynski-NOAA) [inline sorc/gdas.cd:1]: Please check your GDAS App version. I think this is an earlier version than the one in develop.

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: sorc/gdas.cd
Comment (InnocentSouopgui-NOAA) [reply sorc/gdas.cd:1]: That is true. there was no change in GDAS and I did think about updating its hash.
I believe GDAS is not supported on Jet yet.  However, If needed I can update it to the latest in the develop branch.
Let me know.
Let me tag @DavidHuber-NOAA @KateFriedman-NOAA for comments on the support of GDAS App on Jet.

Comment (DavidHuber-NOAA) [reply sorc/gdas.cd:1]: @InnocentSouopgui-NOAA You are correct, the GDASApp is not supported on Jet at this time.  However, this isn't the issue that Walter is raising.  The hash of the GDASApp that you have in `migration-jet-contrib` is older that the hash that is in the global-workflow develop branch.  Can you update your `GDASApp` branch?

```bash
cd global-workflow/sorc/gdas.cd
git checkout faa95efb18f0
cd ../
git add gdas.cd
git commit
git push
```

File: ush/forecast_predet.sh (hunk 1 above)
Comment (InnocentSouopgui-NOAA) [reply ush/forecast_predet.sh]: Done.

File: sorc/gdas.cd
Comment (InnocentSouopgui-NOAA) [reply sorc/gdas.cd:1]: Done.

**Review by WalterKolczynski-NOAA — DISMISSED** [review]

**Comment by DavidHuber-NOAA** [comment]
@InnocentSouopgui-NOAA The gsi-utils bug has been fixed.  You should be able to update the hash now.

**Review by KateFriedman-NOAA — DISMISSED** [review]
Updates look good, thanks @InnocentSouopgui-NOAA ! FYI I am updating the obsproc, prepobs and fit2obs versions in PR #2903 so those external pieces will be taken care of by the time this goes in.

File: sorc/gsi_utils.fd
Comment (DavidHuber-NOAA) [inline sorc/gsi_utils.fd:1]: @InnocentSouopgui-NOAA Can you please update this to the following hash: https://github.com/NOAA-EMC/GSI-utils/commit/a6ea311e5c82369d255e3afdc99c1bce0c9a3014?  This fixes an issue when running JEDI-based analyses.

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]

File: sorc/gsi_utils.fd
Comment (InnocentSouopgui-NOAA) [reply sorc/gsi_utils.fd:1]: Done.

**Review by DavidHuber-NOAA — DISMISSED** [review]
Changes look good.  Thanks for the work @InnocentSouopgui-NOAA!

**Review by WalterKolczynski-NOAA — DISMISSED** [review]

**Review by KateFriedman-NOAA — DISMISSED** [review]
Thanks @InnocentSouopgui-NOAA !

**Review by KateFriedman-NOAA — DISMISSED** [review]
Updates look good, thanks @InnocentSouopgui-NOAA !

**Review by DavidHuber-NOAA — DISMISSED** [review]

**Comment by InnocentSouopgui-NOAA** [comment]
@KateFriedman-NOAA , @DavidHuber-NOAA 
The automated tests failed on wcoss2. 
Can you have a look to investigate further? The error says "CPU oversubscription detected for application".
I do not have access to wcoss2 to check what is going on.

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA , @DavidHuber-NOAA The automated tests failed on wcoss2. Can you have a look to investigate further? The error says "CPU oversubscription detected for application". I do not have access to wcoss2 to check what is going on.

That looks like a similar problem that cropped up in PR #2895 and which @DavidHuber-NOAA had that PR include updates to the prep job resources to resolve it on WCOSS2. Sync your PR branch with g-w `develop` and we'll retry the WCOSS2 CI again with those updated prep job resources.

**Comment by InnocentSouopgui-NOAA** [comment]
> > @KateFriedman-NOAA , @DavidHuber-NOAA The automated tests failed on wcoss2. Can you have a look to investigate further? The error says "CPU oversubscription detected for application". I do not have access to wcoss2 to check what is going on.
> 
> That looks like a similar problem that cropped up in PR #2895 and which @DavidHuber-NOAA had that PR include updates to the prep job resources to resolve it on WCOSS2. Sync your PR branch with g-w `develop` and we'll retry the WCOSS2 CI again with those updated prep job resources.

I just updated my branch.

**Comment by InnocentSouopgui-NOAA** [comment]
@KateFriedman-NOAA , @DavidHuber-NOAA 

WCOSS2 has a disk space issue.

**Disk quota exceeded** in `gfsfcst_seg0.log`
**FATAL ERROR: write error** in `gfsatmos_prod_f###.log`

**Comment by KateFriedman-NOAA** [comment]
Hmmmmmm the stmp quota on WCOSS2-Cactus is currently at 66%. It's possible it hit 100% and then fell overnight because of the scrubber. We'd need to retry the CI test again.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**